cognite/client/_api/iam.py

```diff
@@ -447,14 +448,39 @@ def revoke(self, id: int | Sequence[int]) -> SessionList:
 
         return SessionList._load(self._post(self._RESOURCE_PATH + "/revoke", items).json()["items"])
 
-    def list(self, status: str | None = None) -> SessionList:
+    @overload
+    def retrieve(self, id: int) -> Session: ...
+
+    @overload
+    def retrieve(self, id: Sequence[int]) -> SessionList: ...
+
+    def retrieve(self, id: int | Sequence[int]) -> Session | SessionList:
+        """`Retrieves sessions with given IDs. <https://developer.cognite.com/api#tag/Sessions/operation/getSessionsByIds>`_
+
+        The request will fail if any of the IDs does not belong to an existing session.
+
+        Args:
+            id (int | Sequence[int]): Id or list of session ids
+
+        Returns:
+            Session | SessionList: Session or list of sessions.
+        """
+
+        identifiers = IdentifierSequence.load(ids=id, external_ids=None)
+        return self._retrieve_multiple(
+            list_cls=SessionList,
+            resource_cls=Session,
+            identifiers=identifiers,
+        )
+
+    def list(self, status: str | SessionStatus | None = None) -> SessionList:
         """`List all sessions in the current project. <https://developer.cognite.com/api#tag/Sessions/operation/listSessions>`_
 
         Args:
-            status (str | None): If given, only sessions with the given status are returned.
+            status (str | SessionStatus | None): If given, only sessions with the given status are returned.
```

Comment:
I vote to just rip off the band-aid 😅 
```suggestion
            status (SessionStatus | None): If given, only sessions with the given status are returned.
```